core/unit_test/CMakeLists.txt

```diff
@@ -67,106 +67,149 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;OpenMPTarget;HIP)
+
+    SET(${Tag}_SOURCES2)
+    foreach(Name
+      TeamBasic
+      TeamReductionScan
+      TeamScratch
+      TeamTeamSize
+      TeamVectorRange
+      UniqueToken
+      ViewAPI_a
+      ViewAPI_b
+      ViewAPI_c
+      ViewAPI_d
+      ViewAPI_e
+      ViewLayoutStrideAssignment
+      ViewMapping_a
+      ViewMapping_b
+      ViewMapping_subview
+      ViewOfClass
+      WorkGraph
+      View_64bit
+      ViewResize
+      )
+      set(file ${dir}/Test${Tag}_${Name}.cpp)
+      file(WRITE ${file}
+          "#include <Test${Tag}_Category.hpp>\n"
+          "#include <Test${Name}.hpp>\n"
+      )
+      list(APPEND ${Tag}_SOURCES2 ${file})
+    endforeach()
+
+    set(TagHostAccessible ${Tag})
+    if (Tag STREQUAL "Cuda")
+      set(TagHostAccessible CudaUVM)
+    elseif(Tag STREQUAL "HIP")
+      set(TagHostAccessible HIPHostPinned)
+    endif()
+    foreach(Name
+      SubView_a
+      SubView_b
+      SubView_c01
+      SubView_c02
+      SubView_c03
+      SubView_c04
+      SubView_c05
+      SubView_c06
+      SubView_c07
+      SubView_c08
+      SubView_c09
+      SubView_c10
+      SubView_c11
+      SubView_c12
+      SubView_c13
+      )
+      set(file ${dir}/Test${Tag}_${Name}.cpp)
+      file(WRITE ${file}
+	  "#include <Test${TagHostAccessible}_Category.hpp>\n"
+          "#include <Test${Name}.hpp>\n"
+)
+      list(APPEND ${Tag}_SOURCES2 ${file})
+    endforeach()
 
+    SET(${Tag}_SOURCES ${${Tag}_SOURCES1} ${${Tag}_SOURCES2})
+  endif()
 endforeach()
 
 if(Kokkos_ENABLE_OPENMPTARGET)
   list(REMOVE_ITEM OpenMPTarget_SOURCES
-    openmptarget/TestOpenMPTarget_AtomicOperations_complexfloat.cpp
-    openmptarget/TestOpenMPTarget_AtomicOperations_complexdouble.cpp
-    openmptarget/TestOpenMPTarget_MDRange_a.cpp
-    openmptarget/TestOpenMPTarget_MDRange_b.cpp
-    openmptarget/TestOpenMPTarget_MDRange_c.cpp
-    openmptarget/TestOpenMPTarget_MDRange_d.cpp
-    openmptarget/TestOpenMPTarget_MDRange_e.cpp
-    openmptarget/TestOpenMPTarget_MDRange_f.cpp
-    openmptarget/TestOpenMPTarget_Other.cpp
-    openmptarget/TestOpenMPTarget_Scan.cpp
-    openmptarget/TestOpenMPTarget_Team.cpp
-    openmptarget/TestOpenMPTarget_TeamScratch.cpp
-    openmptarget/TestOpenMPTarget_ViewAPI_e.cpp
-    openmptarget/TestOpenMPTarget_ViewMapping_subview.cpp
-    openmptarget/TestOpenMPTarget_ViewOfClass.cpp
-    openmptarget/TestOpenMPTarget_TeamReductionScan.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_AtomicOperations_complexfloat.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_AtomicOperations_complexdouble.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_Crs.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_LocalDeepCopy.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_MDRange_a.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_MDRange_b.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/openmptarget/TestOpenMPTarget_MDRange_c.cpp
```

Comment:
We're not really changing what was done before so I wouldn't say change this now. But with this refactor we should be able to implement a better way to disable certain tests depending on the backend than just removing certain source files from a list.

Reply:
For me, this is a pretty convenient way of disabling tests but I am happy to find better ways in subsequent pull request.